src/org/labkey/test/util/ArtifactCollector.java

```diff
@@ -161,26 +162,35 @@ public void dumpPageSnapshot(String fileSuffix, @Nullable String subdir, boolean
 
     public void dumpHeap()
     {
-        if (!isLocalServer())
-            return;
-        if ( _test.isGuestModeTest() )
-            return;
-        if (!isHeapDumpCollectionEnabled())
+        if (_dumpedHeap || // Only one heap dump per suite (don't want to overload TeamCity)
+            !isLocalServer() ||
+            _test.isGuestModeTest() ||
+            !isHeapDumpCollectionEnabled()
+        )
+        {
             return;
+        }
 
         _driver.pushLocation();
 
         // Use dumpHeapAction rather that touching file so that we can get file name and publish artifact.
         _driver.beginAt("/admin/dumpHeap.view");
-        File destDir = ensureDumpDir();
         String dumpMsg = Locators.bodyPanel().childTag("div").findElement(_driver.getDriver()).getText();
-        String filename = dumpMsg.substring(dumpMsg.indexOf("HeapDump_"));
-        File heapDump = new File(TestFileUtils.getLabKeyRoot() + "/build/deploy", filename);
-        File destFile = new File(destDir, filename);
+        String filePrefix = "Heap dumped to ";
+        int prefixIndex = dumpMsg.indexOf(filePrefix);
+        if (prefixIndex < 0)
+        {
+            _test.checker().error("Unable to extract filename from page body. Update test or disable heap dump.\n" + dumpMsg);
```

Comment:
This failure message feels a little cryptic to me. I'm not sure I would know what to do if I saw this in a run failure. Specifically I am unclear on what or how I should update the test.